src/lib/mdns/TxtFields.cpp

```diff
@@ -0,0 +1,281 @@
+/*
+ *
+ *    Copyright (c) 2021 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include "TxtFields.h"
+
+#include <algorithm>
+#include <cstdio>
+#include <inttypes.h>
+#include <limits>
+#include <string.h>
+
+#include <mdns/Resolver.h>
+
+namespace chip {
+namespace Mdns {
+
+namespace Internal {
+
+namespace {
+bool IsKey(const ByteSpan & key, const char * desired)
+{
+    if (key.size() != strlen(desired))
+    {
+        return false;
+    }
+    return memcmp(key.data(), desired, key.size()) == 0;
```

Comment:
Per https://datatracker.ietf.org/doc/html/rfc6763#section-6.4 ( Rules for Keys in DNS-SD Key/Value Pairs):

>    Case is ignored when interpreting a key, so "papersize=A4",
>   "PAPERSIZE=A4", and "Papersize=A4" are all identical.

Suggest:

```
char safe_tolower(uint8_t ch)
{
    return static_cast<char>(std::tolower(static_cast<unsigned char>(ch)));
}
.......
    
bool IsKey(const ByteSpan & key, const char * desired)
{
    if (key.size() != strlen(desired))
    {
        return false;
    }

    const uint8_t *cursor = key.data();
    for (size_t idx = 0; idx < key.size(); ++idx)
    {
         // Compare buffer case-insensitively
         if (safe_tolower(*cursor++) != safe_tolower(static_cast<uint8_t>(desired[idx])))
         {
             return false;
         }
        return true;
    }
```

Overall it would be great to have a unit-tested utility for case-insensitive compare for byte buffer and/or string combinations (e.g. possibly using string views), but that's not happening today, so let's get at least this particular one working right.